woocommerce/woocommerce-android
# [Issue 3885] product variation delete confirmation dialog to a dialog fragment

Closes: #3885


### Description
This PR was created to address the issue where the confirmation dialog is dismissed upon device rotation in the `VariationDetailFragment`, some notable changes:

1. Reattaching the Listener: A method named `reattachDialogInteractionListener` was added to the VariationDetailFragment. This method is called within `onViewCreated` to re-establish the connection between the `VariationDetailFragment` and the `WooDialogFragment`. It checks if the dialog is currently displayed and, if so, sets the current instance of `VariationDetailFragment` as the listener to the dialog. This ensures that the dialog can continue to communicate with the active instance of the fragment after a configuration change.

3. Event Handling: The ViewModel was updated to handle the delete action through a `ShowDialogFragment` event. This event encapsulates all necessary information to display the dialog, including message IDs and button texts. When the event is triggered, the dialog is shown with the current context, and any actions taken in the dialog are communicated back to the ViewModel through the established listener interface.

4. Observing Lifecycle Events: The `VariationDetailFragment `observes the `ShowDialogFragment` event. If the fragment is recreated due to an orientation change, the observer ensures that the dialog is shown again with the correct listener attached.

### Testing instructions

1. Click a production with multiple variations. 
2. Click Variations. 
3. Open a variation. 
4. Click the menu option. 
5. Press Delete. 
6. See the dialog visible. 
7. Rotate the device and realize that the dialog is still open. 
8. Verify that the actions on the dialog work as expected.

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/1509205/b8135256-c6e0-4c42-807d-b9887b17b202
